rocAL/source/augmentations/effects_augmentations/node_normalize.cpp

```diff
@@ -0,0 +1,102 @@
+/*
+Copyright (c) 2024 Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#include "augmentations/effects_augmentations/node_normalize.h"
+
+#include <vx_ext_rpp.h>
+
+#include "pipeline/exception.h"
+
+NormalizeNode::NormalizeNode(const std::vector<Tensor *> &inputs, const std::vector<Tensor *> &outputs) : Node(inputs, outputs) {}
+
+void NormalizeNode::create_node() {
+    if (_node)
+        return;
+    unsigned char compute_mean_and_stddev = 0;
+    if (!_mean.size() && !_std_dev.size()) {
+        compute_mean_and_stddev = 3;
```

Comment:
this can be enum. not clear what these 3 cases are for

Reply:
Depending on whether the mean and stddev values are empty or not, we are setting compute_mean_and_stddev value to be passed to rpp kernel. In the rpp kernel, the mean and stddev values will be calculated from the input tensors according to the compute_mean_and_stddev value

Reply:
Please add an enum for this as Lakshmi suggested. Hardcoded values are ambiguous

Reply:
Added an enum in node_normalize.h and used it in the source file for setting the values